src/kw/kw.cpp

```diff
@@ -372,16 +383,72 @@ bool KanoWorld::save_data()
     json_object_set_string(
         json_object(user_data.get()), "token", get_token().c_str());
     json_object_set_string(
-        json_object(user_data.get()), "duration", get_expiration_date().c_str());
+        json_object(user_data.get()), "duration",
+                    get_expiration_date().c_str());
+    json_object_set_boolean(
+        json_object(user_data.get()), "is_verified",
+                    this->is_verified_cache);
 
-    JSON_Status rc = json_serialize_to_file(user_data.get(), data_filename.c_str());
+    JSON_Status rc = json_serialize_to_file(
+        user_data.get(), data_filename.c_str());
 
     return rc == JSONSuccess;
 }
 
+
 /**
- * \warning Not implemented
+ * \warning Implementation not finalised
  */
-bool KanoWorld::is_account_verified() {
-    return true;
+bool KanoWorld::is_account_verified(const bool cache) {
+    this->load_data();
+
+    if (cache && this->is_verified_cache) {
+        return true;
+    }
+
+    this->is_verified_cache = this->is_account_verified_api();
+    this->save_data();
+
+    return this->is_verified_cache;
+}
+
+
+void KanoWorld::clear_account_verified_cache() {
+    this->load_data();
+    this->is_verified_cache = false;
+    this->save_data();
+}
+
+
+/**
+ * \warning Implementation not finalised
+ */
+bool KanoWorld::is_account_verified_api() {
+    std::string auth_token = this->get_token();
+
+    if (auth_token.empty()) {
+        return false;
+    }
+
+    std::map<std::string, std::string> headers{
+            { "Authorization", "Bearer " + auth_token }
+    };
+
+    std::shared_ptr<JSON_Value> response;
+
+    try {
+        response = this->http_client->GET(
+            this->api_url + "/users/me", headers);
+    } catch (const std::exception& e) {
+        return false;
+    }
+
+    const JSON_Object *data = json_value_get_object(response.get());
+    int verified = json_object_dotget_boolean(data, "data.user.isVerified");
```

Comment:
I'd verify this beforehand with `json_object_dothas_value_of_type(data, "data.user.isVerified", JSONBoolean)` and potentially throw an appropriately typed exception about data supplied not being as expected.

Reply:
Checking beforehand. Not throwing in the case that this doesn't exist as this simply means that the server isn't providing the field so the function (whose sole purpose is to tell the user if they have consent or not) can provide the default value. Should probably log in this case.

Reply:
Default value should then match API default. Probably "false"?